python/src/vmaf/tools/ResultAssembler.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import re
+
+from collections import OrderedDict
+from xml.etree import ElementTree
+import json
+import copy, numpy as np
+
+__copyright__ = "Copyright 2016-2018, Netflix, Inc."
+__license__ = "Apache, Version 2.0"
+
+
+def print_usage():
+    print("usage: python " + os.path.basename(sys.argv[0]) + " input_file output_file\n")
+    print("input_file contains a list of files for assembly (can be xml or json)")
+    print("output_file contains the assembled file")
+
+class FileAssembler():
```

Comment:
If these classes go to tools.py, it means they much be very general. However, this is not the case. For example, these classes rely on underlying structure of 'scores', which is specifically related to the Result class. Logically, the assemble functions you are implementing is associated with Result, so they should be moved under Result as 'staticmethod'.

This is a solution, but it may not be the most elegant one. I am thinking what would be the best way to do this. Result already has 'to_json' and 'to_xml' methods, which is to convert a Result object to a file. Similarly, you can create 'from_json' and 'from_xml', which can convert a file to a Result object. Then, to assemble two files, you simply need to write a method to assemble two Result objects into one new Result object. After assembly, you can call 'to_json' or 'to_xml' to convert the new object to a file. This seems to be the most elegant way, since it avoids issues like you have to implement twice the logic associated with calculating the 'aggregate'.